Utilities/WDLs/Dipcall.wdl

```diff
@@ -0,0 +1,270 @@
+        set -o pipefail
+        # cause a bash script to exit immediately when a command fails
+        set -e
+        # cause the bash shell to treat unset variables as an error and exit immediately
+        set -u
+        # echo each line of the script to stdout so we can see what is happening
+        # to turn off echo do 'set +o xtrace'
+        set -o xtrace
+        PATH="/root/bin/samtools_1.9:$PATH"
+
+        # get output base
+        PREFIX=$(basename ~{assemblyFastaPat} | sed 's/.gz$//' | sed 's/.fa\(sta\)*$//' | sed 's/[._][pm]at\(ernal\)*//')
+        mkdir $PREFIX.dipcall
+
+        # prep paternal
+        PAT_FILENAME=$(basename -- "~{assemblyFastaPat}")
+        if [[ $PAT_FILENAME =~ \.gz$ ]]; then
+            cp ~{assemblyFastaPat} .
+            gunzip $PAT_FILENAME
+            PAT_FILENAME="${PAT_FILENAME%.gz}"
+        else
+            ln -s ~{assemblyFastaPat}
+        fi
+
+        # prep maternal
+        MAT_FILENAME=$(basename -- "~{assemblyFastaMat}")
+        if [[ $MAT_FILENAME =~ \.gz$ ]]; then
+            cp ~{assemblyFastaMat} .
+            gunzip $MAT_FILENAME
+            MAT_FILENAME="${MAT_FILENAME%.gz}"
+        else
+            ln -s ~{assemblyFastaMat}
+        fi
+
+        # prep reference
+        REF_FILENAME=$(basename -- "~{referenceFasta}")
+        if [[ $REF_FILENAME =~ \.gz$ ]]; then
+            cp ~{referenceFasta} .
+            gunzip $REF_FILENAME
+            REF_FILENAME="${REF_FILENAME%.gz}"
+        else
+            ln -s ~{referenceFasta}
+        fi
+        samtools faidx $REF_FILENAME
+
+        # initialize script
+        cmd=( /opt/dipcall/dipcall.kit/run-dipcall )
+
+        # male samples need PAR region excluded
+        if [[ ~{isMaleSample} == true ]]; then
+            if [[ ~{referenceIsHS38} ]]; then
+                cmd+=( -x /opt/dipcall/dipcall.kit/hs38.PAR.bed )
+            elif [ -f "~{custom_PAR_bed}" ]; then
+                cmd+=( -x "~{custom_PAR_bed}" )
+            else
+                echo "WARNING: Assuming reference is hg37 because input is not hg38 and no custom PAR provided."
+                echo "Using built-in PAR regions for hg37 to exclude from calling..."
+                cmd+=( -x /opt/dipcall/dipcall.kit/hs37d5.PAR.bed )
+            fi
+
+        fi
+
+        # finalize script
+        cmd+=( $PREFIX.dipcall/$PREFIX )
+        cmd+=( $REF_FILENAME )
+        cmd+=( $PAT_FILENAME )
+        cmd+=( $MAT_FILENAME )
+
+        # generate makefile
+        "${cmd[@]}" >$PREFIX.mak
+
+        # run dipcall
+        make -j 2 -f $PREFIX.mak
+
+        # finalize
+        rm $PREFIX.dipcall/*sam.gz
+        tar czvf $PREFIX.dipcall.tar.gz $PREFIX.dipcall/
+        cp $PREFIX.dipcall/$PREFIX.dip.bed $PREFIX.dipcall.bed
+        cp $PREFIX.dipcall/$PREFIX.dip.vcf.gz $PREFIX.dipcall.vcf.gz
+
+        # cleanup
+        rm $REF_FILENAME
+        rm $MAT_FILENAME
+        rm $PAT_FILENAME
+
+	>>>
```

Comment:
A few things are jumping out at me that are best caught by [ShellCheck](https://github.com/koalaman/shellcheck).  I highly recommend installing that plus [miniwdl](https://github.com/chanzuckerberg/miniwdl), and run `miniwdl check </path/to/wdl>`.  It will examine your WDL and if ShellCheck is installed, it will run your shell script through ShellCheck.  I catch tons of stuff this way, and highly recommend it as a gate for pulls.

Reply:
These are both awesome! Thanks for sharing. They definitely gave some good tips are tweaking the script a bit. I made some suggested changes there.